demo/realtime-img2img/main.py

```diff
@@ -0,0 +1,201 @@
+            user_count = user_data.get_user_count()
+            if args.max_queue_size > 0 and user_count >= args.max_queue_size:
+                print("Server is full")
+                await websocket.send_json(
+                    {"status": "error", "message": "Server is full"}
+                )
+                await websocket.close()
+                return
+            try:
+                user_id = uuid.uuid4()
+                print(f"New user connected: {user_id}")
+
+                await user_data.create_user(user_id, websocket)
+                await websocket.send_json(
+                    {
+                        "status": "connected",
+                        "message": "Connected",
+                        "userId": str(user_id),
+                    }
+                )
+                await websocket.send_json({"status": "send_frame"})
+                await handle_websocket_data(user_id, websocket)
+            except WebSocketDisconnect as e:
+                logging.error(f"WebSocket Error: {e}, {user_id}")
+                traceback.print_exc()
+            finally:
+                print(f"User disconnected: {user_id}")
+                user_data.delete_user(user_id)
+
+        async def handle_websocket_data(user_id: uuid.UUID, websocket: WebSocket):
+            if not user_data.check_user(user_id):
+                return HTTPException(status_code=404, detail="User not found")
+            last_time = time.time()
+            try:
+                while True:
+                    if args.timeout > 0 and time.time() - last_time > args.timeout:
+                        await websocket.send_json(
+                            {
+                                "status": "timeout",
+                                "message": "Your session has ended",
+                                "userId": str(user_id),
+                            }
+                        )
+                        await websocket.close()
+                        return
+                    data = await websocket.receive_json()
+                    if data["status"] != "next_frame":
+                        asyncio.sleep(THROTTLE)
+                        continue
+
+                    params = await websocket.receive_json()
+                    params = pipeline.InputParams(**params)
+                    info = pipeline.Info()
+                    params = SimpleNamespace(**params.dict())
+                    if info.input_mode == "image":
+                        image_data = await websocket.receive_bytes()
+                        if len(image_data) == 0:
+                            await websocket.send_json({"status": "send_frame"})
+                            continue
+                        params.image = bytes_to_pil(image_data)
+                    await user_data.update_data(user_id, params)
+                    await websocket.send_json({"status": "wait"})
+                    await asyncio.sleep(THROTTLE)
```

Comment:
minor nit: I believe this `sleep()` is unnecessary, since `await websocket.receive_json()` should block until the client sends a message to the backend server